ldms/src/core/ldms_xprt.c

```diff
@@ -3701,6 +3701,18 @@ int ldms_xprt_connect_by_name(ldms_t x, const char *host, const char *port,
 	} else {
 		rc = ldms_xprt_connect(x, ai->ai_addr, ai->ai_addrlen, cb, cb_arg);
 	}
+
```

Comment:
Are you specifying a callback function, i.e. the _cb_ argument? If you are, this function is not synchronous. If you want a synchronous connection, then you need to specify NULL for the _cb_ argument.

Reply:
We specify a callback function not NULL. I will check and get back to you.